services/scan/src/simple_interpreter.ts

```diff
@@ -24,14 +28,119 @@ export interface CreateInterpreterOptions {
   readonly testMode: boolean;
 }
 
+export type SheetInterpretation = Scan.SheetInterpretation & {
+  pages: SheetOf<PageInterpretationWithFiles>;
+};
+
 /**
  * Provides a simple interface for interpreting a sheet of ballot paper.
  */
 export interface SimpleInterpreter {
   interpret(
     sheetId: Id,
     sheet: SheetOf<string>
-  ): Promise<Result<SheetOf<PageInterpretationWithFiles>, Error>>;
+  ): Promise<Result<SheetInterpretation, Error>>;
+}
+
+function combinePageInterpretationsForSheet(
+  pages: SheetOf<PageInterpretationWithFiles>
+): Scan.SheetInterpretation {
+  const [front, back] = pages;
+  const frontType = front.interpretation.type;
+  const backType = back.interpretation.type;
+
+  if (
+    frontType === 'InvalidElectionHashPage' ||
+    backType === 'InvalidElectionHashPage'
+  ) {
+    return {
+      type: 'InvalidSheet',
+      reason: 'invalid_election_hash',
+    };
+  }
+
+  if (
+    frontType === 'InvalidTestModePage' ||
+    backType === 'InvalidTestModePage'
+  ) {
+    return {
+      type: 'InvalidSheet',
+      reason: 'invalid_test_mode',
+    };
+  }
+
+  if (
+    frontType === 'InvalidPrecinctPage' ||
+    backType === 'InvalidPrecinctPage'
+  ) {
+    return {
+      type: 'InvalidSheet',
+      reason: 'invalid_precinct',
+    };
+  }
+
+  if (frontType === 'UnreadablePage' || backType === 'UnreadablePage') {
+    return {
+      type: 'InvalidSheet',
+      reason: 'unreadable',
+    };
+  }
+
+  // TODO what does this case actually mean?
```

Comment:
This means we found the QR code that tells us it's a HMPB, but we were not able to actually interpret the page layout.